assets/src/edit-story/app/api/apiProvider.js

```diff
@@ -387,21 +387,23 @@ function APIProvider({ children }) {
     [cdnURL, assetsURL]
   );
 
-  // @todo Add pagination and remove loading all the templates.
-  const getCustomPageTemplates = useCallback(() => {
-    let apiPath = customPageTemplates;
-    const perPage = -1;
-    apiPath = addQueryArgs(apiPath, {
-      context: 'edit',
-      per_page: perPage,
-      page: 1,
-    });
-    return apiFetch({ path: apiPath }).then((response) =>
-      response.map((template) => {
-        return { ...template['story_data'], templateId: template.id };
-      })
-    );
-  }, [customPageTemplates]);
+  const getCustomPageTemplates = useCallback(
+    (page = 1) => {
+      let apiPath = customPageTemplates;
+      const perPage = 100;
+      apiPath = addQueryArgs(apiPath, {
+        context: 'edit',
+        per_page: perPage,
+        page,
+      });
+      return apiFetch({ path: apiPath }).then((response) =>
+        response.map((template) => {
```

Comment:
@spacedmonkey / @swissspidy It looks like we need to be able to access the total pages headers here for pagination (to know when not to try to fetch more pages). Could one of you create a small PR for accepting `_web_stories_envelope` as a param for the page templates? Thoughts?

Reply:
Created a PR here - https://github.com/google/web-stories-wp/pull/7479

Reply:
> Also could consider preloading the REST API response in edit-story.php, maybe it makes a difference.

The preloading topic came up [in a previous PR](https://github.com/google/web-stories-wp/pull/7268#discussion_r617423356) as well with the suggestion being that since it's not necessary for the editor loading then no need to preload it. Currently, it's loaded when the Page templates tab is opened, so by the time the user switches to Saved templates, it would likely be loaded already. Thoughts?

Reply:
Makes sense. No change needed then.